src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.RegularCompilationTracker_Generators.cs

```diff
@@ -111,7 +99,7 @@ private sealed partial class RegularCompilationTracker : ICompilationTracker
             var infos = infosOpt.Value;
 
             // If there are no generated documents, bail out immediately.
-            if (infos.Length == 0)
+            if (creationPolicy == GeneratedDocumentCreationPolicy.Create && infos.Length == 0)
```

Comment:
i think it woudl be worth fleshing out the comment above.  specifically why the new enum check is in here.  (basically, just convey taht, as per the docs, the semantics of .CreateRequired is that we reuse the gen docs from the last run, so we need to now do that below).